"fatal error LNK1120 : 4 unresolved externals" during installation

Hi, I just started with the thd2 team and tried to install catkit2.
The conda environment installation worked fine on my windows machine, however the package installation failed with pip.
I seemed to have several errors : the ones i can identify are : LNK2001 and LNK1120.
Below a screenshot of the last part of the error message.

Does anyone had this issue before or know this problem ?

![Image](https://github.com/user-attachments/assets/1e5288f8-07f7-48c2-b4a5-2b199f4d73e9)

```
(catkit2)
cpaviot@pcgil79 MINGW64 ~/repos/catkit2 (develop)
$ pip install -e .
Obtaining file:///C:/Users/cpaviot/repos/catkit2
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
Building wheels for collected packages: catkit2
  Building editable for catkit2 (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building editable for catkit2 (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [267 lines of output]
      2025-04-11 09:45:15,107 - scikit_build_core - INFO - RUN: C:\Users\cpaviot\miniconda3\envs\catkit2\Library\bin\cmake.EXE -E capabilities
      2025-04-11 09:45:15,123 - scikit_build_core - INFO - CMake version: 3.31.2
      *** scikit-build-core 0.11.1 using CMake 3.31.2 (editable)
      2025-04-11 09:45:15,138 - scikit_build_core - INFO - Build directory: C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build
      *** Configuring CMake...
      2025-04-11 09:45:15,154 - scikit_build_core - WARNING - Can't find a Python library, got libdir=None, ldlibrary=None, multiarch=None, masd=None
      2025-04-11 09:45:15,170 - scikit_build_core - INFO - RUN: C:\Users\cpaviot\miniconda3\envs\catkit2\Library\bin\cmake.EXE -S. -BC:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build -CC:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\CMakeInit.txt -DCMAKE_INSTALL_PREFIX=C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\wheel\platlib
      loading initial cache file C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\CMakeInit.txt
      -- Building for: Visual Studio 17 2022
      -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
      -- The C compiler identification is MSVC 19.43.34809.0
      -- The CXX compiler identification is MSVC 19.43.34809.0
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.43.34808/bin/Hostx64/x64/cl.exe - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.43.34808/bin/Hostx64/x64/cl.exe - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      CMake Deprecation Warning at catkit_core/CMakeLists.txt:1 (cmake_minimum_required):
        Compatibility with CMake < 3.10 will be removed from a future version of
        CMake.

        Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
        to tell CMake that the project requires at least <min> but has been updated
        to work with policies introduced by <max> or earlier.


      -- Found nlohmann_json: C:/Users/cpaviot/miniconda3/envs/catkit2/Library/lib/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found version "3.9.1")
      -- Found Protobuf: C:/Users/cpaviot/miniconda3/envs/catkit2/Library/lib/libprotobuf.lib (found version "5.29.3")
      CMake Warning (dev) at C:/Users/cpaviot/miniconda3/envs/catkit2/Lib/site-packages/pybind11/share/cmake/pybind11/FindPythonLibsNew.cmake:101 (message):
        Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
        are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
        the cmake_policy command to set the policy and suppress this warning, or
        preferably upgrade to using FindPython, either by calling it explicitly
        before pybind11, or by setting PYBIND11_FINDPYTHON ON before pybind11.
      Call Stack (most recent call first):
        C:/Users/cpaviot/miniconda3/envs/catkit2/Lib/site-packages/pybind11/share/cmake/pybind11/pybind11Tools.cmake:50 (find_package)
        C:/Users/cpaviot/miniconda3/envs/catkit2/Lib/site-packages/pybind11/share/cmake/pybind11/pybind11Common.cmake:228 (include)
        C:/Users/cpaviot/miniconda3/envs/catkit2/Lib/site-packages/pybind11/share/cmake/pybind11/pybind11Config.cmake:250 (include)
        catkit2/CMakeLists.txt:10 (find_package)
      This warning is for project developers.  Use -Wno-dev to suppress it.

      -- Found PythonInterp: C:/Users/cpaviot/miniconda3/envs/catkit2/python.exe (found suitable version "3.9.12", minimum required is "3.7")
      -- Found PythonLibs: C:/Users/cpaviot/miniconda3/envs/catkit2/libs/python39.lib
      -- Performing Test HAS_MSVC_GL_LTCG
      -- Performing Test HAS_MSVC_GL_LTCG - Success
      -- Found pybind11: C:/Users/cpaviot/miniconda3/envs/catkit2/Lib/site-packages/pybind11/include (found version "2.13.6")
      -- Found nlohmann_json: C:/Users/cpaviot/miniconda3/envs/catkit2/Library/lib/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found suitable version "3.9.1", minimum required is "3.2.0")
      -- Configuring done (4.7s)
      -- Generating done (0.4s)
      -- Build files have been written to: C:/Users/cpaviot/AppData/Local/Temp/tmp15tlcmk4/build
      *** Building project with Visual Studio 17 2022...
      2025-04-11 09:45:20,290 - scikit_build_core - INFO - RUN: C:\Users\cpaviot\miniconda3\envs\catkit2\Library\bin\cmake.EXE --build C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build -v --config Release
      Change Dir: 'C:/Users/cpaviot/AppData/Local/Temp/tmp15tlcmk4/build'

      Run Build Command(s): "C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/MSBuild/Current/Bin/amd64/MSBuild.exe" ALL_BUILD.vcxproj /p:Configuration=Release /p:Platform=x64 /p:VisualStudioVersion=17.0 /v:n
      MSBuild version 17.13.19+0d9f5a35a for .NET Framework
      Build started 4/11/2025 9:45:20 AM.

      Project "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ALL_BUILD.vcxproj" on node 1 (default targets).
      Project "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ALL_BUILD.vcxproj" (1) is building "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ZERO_CHECK.vcxproj" (2) on node 1 (default targets).
      PrepareForBuild:
        Creating directory "x64\Release\ZERO_CHECK\".
      C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(544,5): warning MSB8029: The Intermediate directory or Output directory cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ZERO_CHECK.vcxproj]
        Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://aka.ms/cpp/structured-output for more details.
        Creating directory "x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\".
      InitializeBuildStatus:
        Creating "x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
        Touching "x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild".
      CustomBuild:
        1>Checking Build System
      FinalizeBuildStatus:
        Deleting file "x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild".
        Touching "x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\ZERO_CHECK.lastbuildstate".
      Done Building Project "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ZERO_CHECK.vcxproj" (default targets).
      Project "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ALL_BUILD.vcxproj" (1) is building "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit2_python_protobuf.vcxproj" (3) on node 1 (default targets).
      PrepareForBuild:
        Creating directory "x64\Release\catkit2_python_protobuf\".
      C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(544,5): warning MSB8029: The Intermediate directory or Output directory cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit2_python_protobuf.vcxproj]
        Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://aka.ms/cpp/structured-output for more details.
        Creating directory "x64\Release\catkit2_python_protobuf\catkit2_.ABCB1A6A.tlog\".
      InitializeBuildStatus:
        Creating "x64\Release\catkit2_python_protobuf\catkit2_.ABCB1A6A.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
        Touching "x64\Release\catkit2_python_protobuf\catkit2_.ABCB1A6A.tlog\unsuccessfulbuild".
      CustomBuild:
        Building Custom Rule C:/Users/cpaviot/repos/catkit2/catkit2/CMakeLists.txt
      FinalizeBuildStatus:
        Deleting file "x64\Release\catkit2_python_protobuf\catkit2_.ABCB1A6A.tlog\unsuccessfulbuild".
        Touching "x64\Release\catkit2_python_protobuf\catkit2_.ABCB1A6A.tlog\catkit2_python_protobuf.lastbuildstate".
      Done Building Project "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit2_python_protobuf.vcxproj" (default targets).
      Project "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ALL_BUILD.vcxproj" (1) is building "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj" (4) on node 1 (default targets).
      Project "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj" (4) is building "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\catkit_core.vcxproj" (5) on node 1 (default targets).
      PrepareForBuild:
        Creating directory "catkit_core.dir\Release\".
      C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(544,5): warning MSB8029: The Intermediate directory or Output directory cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\catkit_core.vcxproj]
        Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://aka.ms/cpp/structured-output for more details.
        Creating directory "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\Release\".
        Creating directory "catkit_core.dir\Release\catkit_core.tlog\".
      InitializeBuildStatus:
        Creating "catkit_core.dir\Release\catkit_core.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
        Touching "catkit_core.dir\Release\catkit_core.tlog\unsuccessfulbuild".
      CustomBuild:
        Running cpp protocol buffer compiler on ../proto/core.proto
        Running cpp protocol buffer compiler on ../proto/logging.proto
        Running cpp protocol buffer compiler on ../proto/service.proto
        Running cpp protocol buffer compiler on ../proto/testbed.proto
        Running cpp protocol buffer compiler on ../proto/tracing.proto
        Building Custom Rule C:/Users/cpaviot/repos/catkit2/catkit_core/CMakeLists.txt
      ClCompile:
        C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.43.34808\bin\HostX64\x64\CL.exe /c /IC:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\gen /nologo /W1 /WX- /diagnostics:column /MP1 /O2 /Ob2 /D _MBCS /D WIN32 /D _WINDOWS /D NDEBUG /D PROTOBUF_USE_DLLS /D "CMAKE_INTDIR=\"Release\"" /EHsc /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /std:c++17 /Fo"catkit_core.dir\Release\\" /Fd"C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\Release\catkit_core.pdb" /external:W0 /Gd /TP /wd4251 /errorReport:queue  /external:I "C:/Users/cpaviot/miniconda3/envs/catkit2/Library/include" /external:I "C:/Users/cpaviot/miniconda3/envs/catkit2/Library/include/eigen3" C:\Users\cpaviot\repos\catkit2\catkit_core\Client.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\Command.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\DataStream.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\Event.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\FreeListAllocator.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\HashMap.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\HostName.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\Log.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\LogConsole.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\LogFile.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\LogForwarder.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\PoolAllocator.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\Property.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\Server.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\Service.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\ServiceProxy.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\ServiceState.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\Shareable.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\SharedMemory.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\Tensor.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\TestbedProxy.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\Timing.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\Tracing.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\Types.cpp C:\Users\cpaviot\repos\catkit2\catkit_core\Util.cpp C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\gen\core.pb.cc C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\gen\logging.pb.cc C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\gen\service.pb.cc C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\gen\testbed.pb.cc C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\gen\tracing.pb.cc
        Client.cpp
        Command.cpp
        DataStream.cpp
        Event.cpp
        FreeListAllocator.cpp
      C:\Users\cpaviot\repos\catkit2\catkit_core\FreeListAllocator.cpp(118,36): warning C4319: '~'ÿ: z‚ro ‚tendant 'uint32_t' vers 'size_t' d'une taille sup‚rieure [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\catkit_core.vcxproj]
        HashMap.cpp
        HostName.cpp
        Log.cpp
        LogConsole.cpp
        LogFile.cpp
        LogForwarder.cpp
        PoolAllocator.cpp
        Property.cpp
        Server.cpp
        Service.cpp
        ServiceProxy.cpp
        ServiceState.cpp
        Shareable.cpp
        SharedMemory.cpp
        Tensor.cpp
        G‚n‚ration de code en cours...
        Compilation en cours...
        TestbedProxy.cpp
        Timing.cpp
        Tracing.cpp
        Types.cpp
        Util.cpp
        core.pb.cc
        logging.pb.cc
        service.pb.cc
        testbed.pb.cc
        tracing.pb.cc
        G‚n‚ration de code en cours...
      Lib:
        C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.43.34808\bin\HostX64\x64\Lib.exe /OUT:"C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\Release\catkit_core.lib" /NOLOGO /MACHINE:X64  /machine:x64 catkit_core.dir\Release\Client.obj
        catkit_core.dir\Release\Command.obj
        catkit_core.dir\Release\DataStream.obj
        catkit_core.dir\Release\Event.obj
        catkit_core.dir\Release\FreeListAllocator.obj
        catkit_core.dir\Release\HashMap.obj
        catkit_core.dir\Release\HostName.obj
        catkit_core.dir\Release\Log.obj
        catkit_core.dir\Release\LogConsole.obj
        catkit_core.dir\Release\LogFile.obj
        catkit_core.dir\Release\LogForwarder.obj
        catkit_core.dir\Release\PoolAllocator.obj
        catkit_core.dir\Release\Property.obj
        catkit_core.dir\Release\Server.obj
        catkit_core.dir\Release\Service.obj
        catkit_core.dir\Release\ServiceProxy.obj
        catkit_core.dir\Release\ServiceState.obj
        catkit_core.dir\Release\Shareable.obj
        catkit_core.dir\Release\SharedMemory.obj
        catkit_core.dir\Release\Tensor.obj
        catkit_core.dir\Release\TestbedProxy.obj
        catkit_core.dir\Release\Timing.obj
        catkit_core.dir\Release\Tracing.obj
        catkit_core.dir\Release\Types.obj
        catkit_core.dir\Release\Util.obj
        catkit_core.dir\Release\core.pb.obj
        catkit_core.dir\Release\logging.pb.obj
        catkit_core.dir\Release\service.pb.obj
        catkit_core.dir\Release\testbed.pb.obj
        catkit_core.dir\Release\tracing.pb.obj
        catkit_core.vcxproj -> C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\Release\catkit_core.lib
      FinalizeBuildStatus:
        Deleting file "catkit_core.dir\Release\catkit_core.tlog\unsuccessfulbuild".
        Touching "catkit_core.dir\Release\catkit_core.tlog\catkit_core.lastbuildstate".
      Done Building Project "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\catkit_core.vcxproj" (default targets).
      PrepareForBuild:
        Creating directory "catkit_bindings.dir\Release\".
      C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(544,5): warning MSB8029: The Intermediate directory or Output directory cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
        Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://aka.ms/cpp/structured-output for more details.
        Creating directory "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\Release\".
        Creating directory "catkit_bindings.dir\Release\catkit_bindings.tlog\".
      InitializeBuildStatus:
        Creating "catkit_bindings.dir\Release\catkit_bindings.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
        Touching "catkit_bindings.dir\Release\catkit_bindings.tlog\unsuccessfulbuild".
      CustomBuild:
        Building Custom Rule C:/Users/cpaviot/repos/catkit2/catkit2/CMakeLists.txt
      ClCompile:
        C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.43.34808\bin\HostX64\x64\CL.exe /c /IC:\Users\cpaviot\repos\catkit2\catkit2\pybind11_json_INCLUDE_DIRS /IC:\Users\cpaviot\repos\catkit2\catkit2\..\catkit_core /IC:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\gen /nologo /W1 /WX- /diagnostics:column /MP /O2 /Ob2 /GL /D _WINDLL /D _MBCS /D WIN32 /D _WINDOWS /D NDEBUG /D PROTOBUF_USE_DLLS /D "CMAKE_INTDIR=\"Release\"" /D catkit_bindings_EXPORTS /EHsc /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /std:c++17 /Fo"catkit_bindings.dir\Release\\" /Fd"catkit_bindings.dir\Release\vc143.pdb" /external:W0 /Gd /TP /wd4251 /errorReport:queue  /external:I "C:/Users/cpaviot/miniconda3/envs/catkit2/Lib/site-packages/pybind11/include" /external:I "C:/Users/cpaviot/miniconda3/envs/catkit2/include" /external:I "C:/Users/cpaviot/miniconda3/envs/catkit2/Library/include" /external:I "C:/Users/cpaviot/miniconda3/envs/catkit2/Library/include/eigen3" /bigobj C:\Users\cpaviot\repos\catkit2\catkit2\bindings.cpp
        bindings.cpp
      Link:
        C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.43.34808\bin\HostX64\x64\link.exe /ERRORREPORT:QUEUE /OUT:"C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\Release\catkit_bindings.cp39-win_amd64.pyd" /INCREMENTAL:NO /NOLOGO ..\catkit_core\Release\catkit_core.lib C:\Users\cpaviot\miniconda3\envs\catkit2\libs\python39.lib "C:\Users\cpaviot\miniconda3\envs\catkit2\Library\lib\libzmq-mt-4_3_4.lib" C:\Users\cpaviot\miniconda3\envs\catkit2\Library\lib\libsodium.lib rpcrt4.lib iphlpapi.lib wsock32.lib ws2_32.lib Iphlpapi.lib C:\Users\cpaviot\miniconda3\envs\catkit2\Library\lib\libprotobuf.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /PDB:"C:/Users/cpaviot/AppData/Local/Temp/tmp15tlcmk4/build/catkit2/Release/catkit_bindings.pdb" /SUBSYSTEM:CONSOLE /LTCG /LTCGOUT:"catkit_bindings.dir\Release\catkit_bindings.cp39-win_amd64.iobj" /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"C:/Users/cpaviot/AppData/Local/Temp/tmp15tlcmk4/build/catkit2/Release/catkit_bindings.lib" /MACHINE:X64  /machine:x64 /DLL catkit_bindings.dir\Release\bindings.obj
           Cr‚ation de la bibliothŠque C:/Users/cpaviot/AppData/Local/Temp/tmp15tlcmk4/build/catkit2/Release/catkit_bindings.lib et de l'objet C:/Users/cpaviot/AppData/Local/Temp/tmp15tlcmk4/build/catkit2/Release/catkit_bindings.exp
      catkit_core.lib(Tensor.obj) : error LNK2001: symbole externe non r‚solu "public: class absl::lts_20250127::log_internal::LogMessage & __cdecl absl::lts_20250127::log_internal::LogMessage::operator<<<int,0>(int const &)" (??$?6H$0A@@LogMessage@log_internal@lts_20250127@absl@@QEAAAEAV0123@AEBH@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
      catkit_core.lib(core.pb.obj) : error LNK2001: symbole externe non r‚solu "public: class absl::lts_20250127::log_internal::LogMessage & __cdecl absl::lts_20250127::log_internal::LogMessage::operator<<<int,0>(int const &)" (??$?6H$0A@@LogMessage@log_internal@lts_20250127@absl@@QEAAAEAV0123@AEBH@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
      catkit_core.lib(Tensor.obj) : error LNK2001: symbole externe non r‚solu "private: void __cdecl absl::lts_20250127::log_internal::LogMessage::CopyToEncodedBuffer<0>(class std::basic_string_view<char,struct std::char_traits<char> >)" (??$CopyToEncodedBuffer@$0A@@LogMessage@log_internal@lts_20250127@absl@@AEAAXV?$basic_string_view@DU?$char_traits@D@std@@@std@@@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
      catkit_core.lib(tracing.pb.obj) : error LNK2001: symbole externe non r‚solu "private: void __cdecl absl::lts_20250127::log_internal::LogMessage::CopyToEncodedBuffer<0>(class std::basic_string_view<char,struct std::char_traits<char> >)" (??$CopyToEncodedBuffer@$0A@@LogMessage@log_internal@lts_20250127@absl@@AEAAXV?$basic_string_view@DU?$char_traits@D@std@@@std@@@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
      catkit_core.lib(core.pb.obj) : error LNK2001: symbole externe non r‚solu "private: void __cdecl absl::lts_20250127::log_internal::LogMessage::CopyToEncodedBuffer<0>(class std::basic_string_view<char,struct std::char_traits<char> >)" (??$CopyToEncodedBuffer@$0A@@LogMessage@log_internal@lts_20250127@absl@@AEAAXV?$basic_string_view@DU?$char_traits@D@std@@@std@@@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
      catkit_core.lib(testbed.pb.obj) : error LNK2001: symbole externe non r‚solu "private: void __cdecl absl::lts_20250127::log_internal::LogMessage::CopyToEncodedBuffer<0>(class std::basic_string_view<char,struct std::char_traits<char> >)" (??$CopyToEncodedBuffer@$0A@@LogMessage@log_internal@lts_20250127@absl@@AEAAXV?$basic_string_view@DU?$char_traits@D@std@@@std@@@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
      catkit_core.lib(Tensor.obj) : error LNK2001: symbole externe non r‚solu "char const * __cdecl absl::lts_20250127::log_internal::MakeCheckOpString<unsigned __int64,unsigned __int64>(unsigned __int64,unsigned __int64,char const *)" (??$MakeCheckOpString@_K_K@log_internal@lts_20250127@absl@@YAPEBD_K0PEBD@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
      catkit_core.lib(core.pb.obj) : error LNK2001: symbole externe non r‚solu "char const * __cdecl absl::lts_20250127::log_internal::MakeCheckOpString<unsigned __int64,unsigned __int64>(unsigned __int64,unsigned __int64,char const *)" (??$MakeCheckOpString@_K_K@log_internal@lts_20250127@absl@@YAPEBD_K0PEBD@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
      catkit_core.lib(Types.obj) : error LNK2001: symbole externe non r‚solu "public: __cdecl absl::lts_20250127::log_internal::LogMessageFatal::~LogMessageFatal(void)" (??1LogMessageFatal@log_internal@lts_20250127@absl@@QEAA@XZ) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
      C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\Release\catkit_bindings.cp39-win_amd64.pyd : fatal error LNK1120: 4 externes non r‚solus [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
      Done Building Project "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj" (default targets) -- FAILED.
      Done Building Project "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ALL_BUILD.vcxproj" (default targets) -- FAILED.

      Build FAILED.

      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ALL_BUILD.vcxproj" (default target) (1) ->
      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ZERO_CHECK.vcxproj" (default target) (2) ->
      (PrepareForBuild target) ->
        C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(544,5): warning MSB8029: The Intermediate directory or Output directory cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ZERO_CHECK.vcxproj]


      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ALL_BUILD.vcxproj" (default target) (1) ->
      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit2_python_protobuf.vcxproj" (default target) (3) ->
        C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(544,5): warning MSB8029: The Intermediate directory or Output directory cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit2_python_protobuf.vcxproj]


      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ALL_BUILD.vcxproj" (default target) (1) ->
      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj" (default target) (4) ->
      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\catkit_core.vcxproj" (default target) (5) ->
        C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(544,5): warning MSB8029: The Intermediate directory or Output directory cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\catkit_core.vcxproj]


      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ALL_BUILD.vcxproj" (default target) (1) ->
      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj" (default target) (4) ->
      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\catkit_core.vcxproj" (default target) (5) ->
      (ClCompile target) ->
        C:\Users\cpaviot\repos\catkit2\catkit_core\FreeListAllocator.cpp(118,36): warning C4319: '~'ÿ: z‚ro ‚tendant 'uint32_t' vers 'size_t' d'une taille sup‚rieure [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit_core\catkit_core.vcxproj]


      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ALL_BUILD.vcxproj" (default target) (1) ->
      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj" (default target) (4) ->
      (PrepareForBuild target) ->
        C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(544,5): warning MSB8029: The Intermediate directory or Output directory cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]


      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\ALL_BUILD.vcxproj" (default target) (1) ->
      "C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj" (default target) (4) ->
      (Link target) ->
        catkit_core.lib(Tensor.obj) : error LNK2001: symbole externe non r‚solu "public: class absl::lts_20250127::log_internal::LogMessage & __cdecl absl::lts_20250127::log_internal::LogMessage::operator<<<int,0>(int const &)" (??$?6H$0A@@LogMessage@log_internal@lts_20250127@absl@@QEAAAEAV0123@AEBH@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
        catkit_core.lib(core.pb.obj) : error LNK2001: symbole externe non r‚solu "public: class absl::lts_20250127::log_internal::LogMessage & __cdecl absl::lts_20250127::log_internal::LogMessage::operator<<<int,0>(int const &)" (??$?6H$0A@@LogMessage@log_internal@lts_20250127@absl@@QEAAAEAV0123@AEBH@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
        catkit_core.lib(Tensor.obj) : error LNK2001: symbole externe non r‚solu "private: void __cdecl absl::lts_20250127::log_internal::LogMessage::CopyToEncodedBuffer<0>(class std::basic_string_view<char,struct std::char_traits<char> >)" (??$CopyToEncodedBuffer@$0A@@LogMessage@log_internal@lts_20250127@absl@@AEAAXV?$basic_string_view@DU?$char_traits@D@std@@@std@@@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
        catkit_core.lib(tracing.pb.obj) : error LNK2001: symbole externe non r‚solu "private: void __cdecl absl::lts_20250127::log_internal::LogMessage::CopyToEncodedBuffer<0>(class std::basic_string_view<char,struct std::char_traits<char> >)" (??$CopyToEncodedBuffer@$0A@@LogMessage@log_internal@lts_20250127@absl@@AEAAXV?$basic_string_view@DU?$char_traits@D@std@@@std@@@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
        catkit_core.lib(core.pb.obj) : error LNK2001: symbole externe non r‚solu "private: void __cdecl absl::lts_20250127::log_internal::LogMessage::CopyToEncodedBuffer<0>(class std::basic_string_view<char,struct std::char_traits<char> >)" (??$CopyToEncodedBuffer@$0A@@LogMessage@log_internal@lts_20250127@absl@@AEAAXV?$basic_string_view@DU?$char_traits@D@std@@@std@@@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
        catkit_core.lib(testbed.pb.obj) : error LNK2001: symbole externe non r‚solu "private: void __cdecl absl::lts_20250127::log_internal::LogMessage::CopyToEncodedBuffer<0>(class std::basic_string_view<char,struct std::char_traits<char> >)" (??$CopyToEncodedBuffer@$0A@@LogMessage@log_internal@lts_20250127@absl@@AEAAXV?$basic_string_view@DU?$char_traits@D@std@@@std@@@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
        catkit_core.lib(Tensor.obj) : error LNK2001: symbole externe non r‚solu "char const * __cdecl absl::lts_20250127::log_internal::MakeCheckOpString<unsigned __int64,unsigned __int64>(unsigned __int64,unsigned __int64,char const *)" (??$MakeCheckOpString@_K_K@log_internal@lts_20250127@absl@@YAPEBD_K0PEBD@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
        catkit_core.lib(core.pb.obj) : error LNK2001: symbole externe non r‚solu "char const * __cdecl absl::lts_20250127::log_internal::MakeCheckOpString<unsigned __int64,unsigned __int64>(unsigned __int64,unsigned __int64,char const *)" (??$MakeCheckOpString@_K_K@log_internal@lts_20250127@absl@@YAPEBD_K0PEBD@Z) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
        catkit_core.lib(Types.obj) : error LNK2001: symbole externe non r‚solu "public: __cdecl absl::lts_20250127::log_internal::LogMessageFatal::~LogMessageFatal(void)" (??1LogMessageFatal@log_internal@lts_20250127@absl@@QEAA@XZ) [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]
        C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\Release\catkit_bindings.cp39-win_amd64.pyd : fatal error LNK1120: 4 externes non r‚solus [C:\Users\cpaviot\AppData\Local\Temp\tmp15tlcmk4\build\catkit2\catkit_bindings.vcxproj]

          5 Warning(s)
          10 Error(s)

      Time Elapsed 00:00:49.96


      *** CMake build failed
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building editable for catkit2
Failed to build catkit2
ERROR: Failed to build installable wheels for some pyproject.toml based projects (catkit2)
(catkit2)
cpaviot@pcgil79 MINGW64 ~/repos/catkit2 (develop)
$
```

--- a/environment.yml
+++ b/environment.yml
@@ -24,8 +24,8 @@ dependencies:
   - conda-forge::asdf
   - pyyaml
   - psutil
-  - protobuf
-  - libprotobuf
+  - protobuf==3.20.3
+  - libprotobuf==3.20.3
   - conda-forge::hcipy
   - conda-forge::cppzmq==4.8.1
   - conda-forge::pybind11==2.13.6

